airbyte-cdk/python/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py

```diff
@@ -30,34 +32,32 @@ class SimpleRetriever(Retriever, HttpStream):
     This retriever currently inherits from HttpStream to reuse the request submission and pagination machinery.
     As a result, some of the parameters passed to some methods are unused.
     The two will be decoupled in a future release.
+
+    Attributes:
+        stream_name (str): The stream's name
+        stream_primary_key (str): The stream's primary key
+        requester (Requester): The HTTP requester
+        record_selector (HttpSelector): The record selector
+        paginator (Optional[Paginator]): The paginator
+        stream_slicer (Optional[StreamSlicer]): The stream slicer
+        options (Mapping[str, Any]): Additional runtime parameters to be used for string interpolation
     """
 
-    def __init__(
-        self,
-        name,
-        primary_key,
-        requester: Requester,
-        record_selector: HttpSelector,
-        paginator: Optional[Paginator] = None,
-        stream_slicer: Optional[StreamSlicer] = SingleSlice(),
-        **options: Optional[Mapping[str, Any]],
-    ):
-        """
-        :param name: The stream's name
-        :param primary_key: The stream's primary key
-        :param requester: The HTTP requester
-        :param record_selector: The record selector
-        :param paginator: The paginator
-        :param stream_slicer: The stream slicer
-        :param options: Additional runtime parameters to be used for string interpolation
-        """
-        self._name = name
-        self._primary_key = primary_key
-        self._paginator = paginator or NoPagination()
-        self._requester = requester
-        self._record_selector = record_selector
-        super().__init__(self._requester.get_authenticator())
-        self._stream_slicer = stream_slicer
+    requester: Requester
+    record_selector: HttpSelector
+    stream_name: str
```

Comment:
Language change: to avoid a name collision with the existing `name()` and `primary_key()` methods that must be implemented as part of `HttpStream` we can't use these as field names.